core-bundle/src/Resources/contao/library/Contao/Database/Statement.php

```diff
@@ -270,8 +263,44 @@ public function query($strQuery='')
 			throw new \Exception('Empty query string');
 		}
 
-		// Execute the query
-		$this->statement = $this->resConnection->executeQuery($this->strQuery);
+		$arrParams = array_map(
+			static function ($varParam)
+			{
+				if (
+					!\is_string($varParam)
+					&& !\is_bool($varParam)
+					&& !\is_float($varParam)
+					&& !\is_int($varParam)
+					&& $varParam !== null
+				) {
+					return serialize($varParam);
+				}
+
+				return $varParam;
```

Comment:
Maybe better comprehensible when de-morganified? :slightly_smiling_face: 
```suggestion
				if (
					\is_string($varParam)
					|| \is_bool($varParam)
					|| \is_float($varParam)
					|| \is_int($varParam)
					|| $varParam === null
				) {
					return $varParam;
				}

				return serialize($varParam);
```